backend/deepchecks_monitoring/api/v1/configuration.py

```diff
@@ -151,7 +171,81 @@ async def get_all_alert_rules(
                 alert_rule_schema.user = user_schema
         alert_rule_schemas.append(alert_rule_schema)
-    return alert_rule_schemas
+
+    # nearly duplicated code (not sure if I can do it preittier)
+    non_resolved_alerts_count = (
+        select(
+            DataIngestionAlert.alert_rule_id.label("alert_rule_id"),
+            func.count(DataIngestionAlert.id).label("non_resolved_alerts"),
+        )
+        .where(DataIngestionAlert.resolved.is_(False))
+        .group_by(DataIngestionAlert.alert_rule_id)
+    )
+
+    non_resolved_alerts_count = non_resolved_alerts_count.subquery()
+
+    total_count = (
+        select(
+            DataIngestionAlert.alert_rule_id.label("alert_rule_id"),
+            func.count(DataIngestionAlert.id).label("total_alerts"),
+            func.max(DataIngestionAlert.end_time).label("recent_alert")
+        )
+        .group_by(DataIngestionAlert.alert_rule_id)
+    )
+
+    total_count = total_count.subquery()
+    severity_index = DataIngestionAlertRule.alert_severity_index.label("severity_index")
+
+    q = (
+        select(
+            DataIngestionAlertRule.id,
+            DataIngestionAlertRule.created_by,
+            DataIngestionAlertRule.condition,
+            DataIngestionAlertRule.name,
+            DataIngestionAlertRule.alert_severity,
+            DataIngestionAlertRule.frequency,
+            DataIngestionAlertRule.alert_type,
+            non_resolved_alerts_count.c.non_resolved_alerts,
+            total_count.c.total_alerts,
+            total_count.c.recent_alert,
+            severity_index
+        )
+        .join(DataIngestionAlertRule.model)
+        .join(Model.members)
+        .where(ModelMember.user_id == user.id)
+        .outerjoin(non_resolved_alerts_count, non_resolved_alerts_count.c.alert_rule_id == DataIngestionAlertRule.id)
+        .outerjoin(total_count, total_count.c.alert_rule_id == DataIngestionAlertRule.id)
+    )
+    if models:
+        q = q.where(DataIngestionAlertRule.model_id.in_(models))
+    if severity:
+        q = q.where(DataIngestionAlertRule.alert_severity.in_(severity))
+
+    if not sortby:
+        q = q.order_by(severity_index.desc(), total_count.c.recent_alert.desc())
+    else:
+        if "severity:asc" in sortby:
+            q = q.order_by(severity_index.asc())
+        if "severity:desc" in sortby:
+            q = q.order_by(severity_index.desc())
+        if "alert-window:asc" in sortby:
+            q = q.order_by(total_count.c.recent_alert.asc())
+        if "alert-window:desc" in sortby:
+            q = q.order_by(total_count.c.recent_alert.desc())
+
+    alert_rules_rows = (await session.execute(q)).all()
+    data_alert_rule_schemas = []
+    for row in alert_rules_rows:
+        alert_rule_schema = DataAlertRuleConfigSchema.from_orm(row)
+        if row.created_by != 0:
+            q = select(User).where(User.id == row.created_by)
+            user = (await session.execute(q)).scalars().first()
```

Comment:
issuing SQL queries within a loop in my opinion is not a very good idea, 
Did you try to `join` rules creators instead of selecting them sequentially?
which solution will have a better performance? (maybe create a separate issue for this)

Reply:
https://linear.app/deepchecks/issue/MON-2372/test-sql-execute-in-loops